app/ante/fee_checker.go

```diff
@@ -11,41 +16,38 @@ const (
 	priorityScalingFactor = 1_000_000
 )
 
-// checkTxFeeWithValidatorMinGasPrices implements the default fee logic, where the minimum price per
-// unit of gas is fixed and set by each validator, and the tx priority is computed from the gas price.
-func checkTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
+// CheckTxFeeWithGlobalMinGasPrices implements the default fee logic, where the minimum price per
+// unit of gas is fixed and set globally, and the tx priority is computed from the gas price.
+func CheckTxFeeWithGlobalMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
 	feeTx, ok := tx.(sdk.FeeTx)
 	if !ok {
 		return nil, 0, errors.Wrap(sdkerror.ErrTxDecode, "Tx must be a FeeTx")
 	}
 
-	feeCoins := feeTx.GetFee()
+	fee := feeTx.GetFee().AmountOf(appconsts.BondDenom)
 	gas := feeTx.GetGas()
 
-	// Ensure that the provided fees meet a minimum threshold for the validator,
-	// if this is a CheckTx. This is only for local mempool purposes, and thus
-	// is only ran on check tx.
-	if ctx.IsCheckTx() {
-		minGasPrices := ctx.MinGasPrices()
-		if !minGasPrices.IsZero() {
-			requiredFees := make(sdk.Coins, len(minGasPrices))
+	// global minimum fee only applies to app versions greater than one
+	if ctx.BlockHeader().Version.App > v1.Version {
+		// check for overflow, gas is now capped at MaxInt64
+		if gas > math.MaxInt64 {
+			return nil, 0, errors.Wrap(sdkerror.ErrInvalidGasLimit, "overflow: gas exceeds MaxInt64")
+		}
```

Comment:
It looks like we actually don't need this check.

We can go:

`gasInt  := sdk.NewIntFromUint64(gas)`

and then go `globalMinGasPrice.MulInt(gasInt).TruncateInt`